### PR TITLE
Long text in the news feed title will be clamped by 3 lines

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@sofiiakvasnevska 

## Issue
https://github.com/Fliplet/fliplet-studio/issues/5635

## Description
Long text in the news feed title will be clamped by 3 lines

## Screenshots/screencasts
image

image

## Backward compatibility

This change is fully backward compatible.

## Reviewers 
@upplabs-alex-levchenko @MaksymShokin

### CODE REVIEW

**Comment by tonytlwu** [comment]
@YaroslavOvdii The issue didn't specify that the titles need to be clamped to 3 lines. It specified that long words need to be capped. That's all.

**Comment by YaroslavOvdii** [comment]
@tonytlwu To clarify, we doesn't need to clamp title at all, for example when our title will be this long: 

>   Lorem ipsum dolor sit amet consectetur adipisicing elit. Doloribus veritatis, debitis quasi perspiciatis quaerat itaque ut vero esse, illo aperiam, repudiandae soluta a pariatur ipsa reprehenderit officiis? Ipsa, obcaecati dolorem.
  Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quam voluptatem obcaecati, delectus repudiandae ipsum mollitia, ipsam dolore odio ut at, deserunt officia repellat? Dolores harum, maiores ipsa iure dicta qui.
  Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptatum accusamus libero quis reiciendis nostrum dignissimos, doloremque sed fuga atque molestiae nihil fugiat ullam sapiente harum repellendus in culpa? Velit, sunt!

Or we don't clamp it exactly to 3 lines?

And if we capped only long words, that is mean that we need to find in text word long enough so it would be out of the box or we can leave only one line for the title and use `text-overflow` on it?

**Comment by tonytlwu** [comment]
@YaroslavOvdii Thanks for explaining. In this case my initial assessment was wrong. We should end long words with ellipsis. You're right that ellipses only make sense if we're also line limiting.

In this case, we should implement https://css-tricks.com/snippets/css/prevent-long-urls-from-breaking-out-of-container/ and break long words.

**Comment by dimitrianoudi** [comment]
Can you make sure that i am not subscribed to any of your repos please?

**Comment by YaroslavOvdii** [comment]
@tonytlwu We should go with 3 lines as for description or another number of lines?

**Comment by tonytlwu** [comment]
@dimitrianoudi Good to hear from you. You're not currently in any of the listed Fliplet teams, which means you're not enforced to receive any notifications from our repos. You will need to unwatch or unfollow this from your end, e.g. https://github.com/Fliplet/fliplet-widget-dynamic-lists/watchers shows that you're currently watching this repo. If you think there's anything else we can and should do, let us know.

**Comment by tonytlwu** [comment]
@YaroslavOvdii See my comment https://github.com/Fliplet/fliplet-widget-dynamic-lists/pull/355#issuecomment-608417281 which says that there should be no limit on the number of lines. The solution should not be dependent on line limiting at all. The comment contains CSS snippets for resolving this.

**Comment by YaroslavOvdii** [comment]
@tonytlwu Do I get it right, that it should look like this?
image

**Comment by tonytlwu** [comment]
@YaroslavOvdii That's correct

**Comment by YaroslavOvdii** [comment]
@tonytlwu 
Used `word-break: break-all;` because of the `word-break: break-word;` doesn't support by IE11.
Current view of the title:
Chrome
image

IE11
image

**Comment by tonytlwu** [comment]
@YaroslavOvdii No, we can't be breaking a word like **capture** here.

According to the article I sent you, you should be able to support it by using all these properties at the same time:

```
  word-break: break-word;
  word-wrap: break-word;
  overflow-wrap: break-word;
```

I don't see you using all of them in your changes. Is there a reason why? Does it help if you add all 3?

**Comment by YaroslavOvdii** [comment]
@tonytlwu Yes, it helped.
The current view is:
Chrome
image

IE11
image

Edge18
image

Firefox
image

File: js/layout-javascript/news-feed-code.js [hunk 1]
```diff
@@ -1435,7 +1435,11 @@ DynamicList.prototype.renderLoopHTML = function () {
           var selector = $lineClamp.length ? '[data-line-clamp]' : '.news-feed-item-description';
           var linesToClamp = $lineClamp.length ? $lineClamp.data('line-clamp') : 3;
 
-          window.ellipsed.ellipsis(selector, linesToClamp);
+          window.ellipsed.ellipsis(selector, linesToClamp, {delimiter: '',});
+        } else {
+          var newsDescriptionSelector = '.news-feed-item-description';
+
+          $(newsDescriptionSelector).css('max-height', '4.4em');
```
Comment (tonytlwu) [inline js/layout-javascript/news-feed-code.js:1442]: Why is this necessary?

File: css/layout-css/news-feed-style.upload.css [hunk 2]
```diff
@@ -761,6 +761,9 @@
   position: relative;
   margin: 8px 0 30px 0;
   transition: all 0.1s;
+  word-break: break-word;
```
Comment (tonytlwu) [inline css/layout-css/news-feed-style.upload.css:764]: Can you also check that long words are correctly broken in all LFD layouts? Please check all the fields that are displayed, e.g. title, description, categories and other labels etc.

File: js/layout-javascript/news-feed-code.js (hunk 1 above)
Comment (YaroslavOvdii) [reply js/layout-javascript/news-feed-code.js:1442]: @tonytlwu 
While I was fixing this issue I've noticed that in IE11 text wasn't ended with ellipses, it was because of the height attribute in the CSS, so I removed it from the CSS. But height is needed for the CSS clamp that we use in other browsers except for the IE11 that is why we are adding it here.

Comment (tonytlwu) [reply js/layout-javascript/news-feed-code.js:1442]: @YaroslavOvdii OK. If it's not related to this issue of long words, then let's keep it separate. Log a different issue and have it go through the same QA process. We've already see in this comment https://github.com/Fliplet/fliplet-studio/issues/5868#issuecomment-610783432 that mixing fixes could lead to confusions in the future.

File: css/layout-css/news-feed-style.upload.css (hunk 2 above)
Comment (YaroslavOvdii) [reply css/layout-css/news-feed-style.upload.css:764]: @tonytlwu I've checked layouts and seems only in the directory long text is working okay.
directory long text
featured list
news feed long text
simple list long text
agenda long text

Should we fix all layouts in this PR?

Comment (tonytlwu) [reply css/layout-css/news-feed-style.upload.css:764]: Yes. Please fix them all.

Comment (YaroslavOvdii) [reply css/layout-css/news-feed-style.upload.css:767]: @tonytlwu 
corrected all layouts how they look now https://share.getcloudapp.com/NQuDEgzX

File: js/layout-javascript/simple-list-code.js [hunk 3]
```diff
@@ -1220,7 +1220,7 @@ DynamicList.prototype.renderLoopHTML = function () {
           var selector = $lineClamp.length ? '[data-line-clamp]' : '.list-item-description';
           var linesToClamp = $lineClamp.length ? $lineClamp.data('line-clamp') : 3;
 
-          window.ellipsed.ellipsis(selector, linesToClamp);
+          window.ellipsed.ellipsis(selector, linesToClamp, {delimiter: ''});
```
Comment (tonytlwu) [inline js/layout-javascript/simple-list-code.js:1223]: @YaroslavOvdii Why is this necessary?

```suggestion
          window.ellipsed.ellipsis(selector, linesToClamp, {delimiter: ''});
```

File: js/layout-javascript/news-feed-code.js [hunk 4]
```diff
@@ -1435,7 +1435,7 @@ DynamicList.prototype.renderLoopHTML = function () {
           var selector = $lineClamp.length ? '[data-line-clamp]' : '.news-feed-item-description';
           var linesToClamp = $lineClamp.length ? $lineClamp.data('line-clamp') : 3;
 
-          window.ellipsed.ellipsis(selector, linesToClamp);
+          window.ellipsed.ellipsis(selector, linesToClamp, {delimiter: '',});
```
Comment (tonytlwu) [inline js/layout-javascript/news-feed-code.js:1438]: @upplabs-alex-levchenko @MaksymShokin 

1. Objects defined in a single line should always have spaces inside the `{}` brackets.
1. No trailing comma in objects or arrays

```suggestion
          window.ellipsed.ellipsis(selector, linesToClamp, { delimiter: '' });
```

File: js/layout-javascript/simple-list-code.js (hunk 3 above)
Comment (YaroslavOvdii) [reply js/layout-javascript/simple-list-code.js:1223]: @tonytlwu  this is needed to clamp long word correctly.
[Docs](https://github.com/nzambello/ellipsed#options).
